src/resolvers/libsodium.rs

```diff
@@ -0,0 +1,280 @@
+extern crate sodiumoxide;
+
+use byteorder::{ByteOrder, LittleEndian};
+
+use params::{CipherChoice, DHChoice, HashChoice};
+use types::{Cipher, Dh, Hash, Random};
+use CryptoResolver;
+
+use self::sodiumoxide::crypto::aead::chacha20poly1305_ietf as sodium_chacha20poly1305;
+use self::sodiumoxide::crypto::hash::sha256 as sodium_sha256;
+use self::sodiumoxide::crypto::scalarmult::curve25519 as sodium_curve25519;
+
+#[derive(Default)]
+pub struct SodiumResolver;
+
+impl CryptoResolver for SodiumResolver {
+    fn resolve_rng(&self) -> Option<Box<Random + Send>> {
+        None
```

Comment:
Any reason we don't want the included RNG in sodiumoxide?

Reply:
Hm... I missed somehow that `sodiumoxide` provides a public interface for that. I'll add it soon then.